src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -190,7 +190,7 @@ private long internalCheck(EC2Computer computer) {
                 }
             }
 
-            final long idleMilliseconds = this.clock.millis() - computer.getIdleStartMilliseconds();
+            final long idleMilliseconds = this.clock.millis() - Math.max(computer.getIdleStartMilliseconds(), uptime);
```

Comment:
So this looks wrong because,
https://github.com/jenkinsci/ec2-plugin/blob/8867fe65acaa999c52dc4689192a8cd5da136234/src/main/java/hudson/plugins/ec2/EC2Computer.java#L146-L148

uptime is `Currtime - instance_launch_time`
idlestartmiliseconds is `instance_idle_start_time`

so wouldnt this be `Currtime - (currtime - instance_launch_time) = instance_launch_time`?

Would the right solution here not be

`this.clock.millis() - Math.max(computer.getIdleStartMilliseconds(), computer.describeInstance().getLaunchTime().getTime());`

instead so we get the time the instance launched?

Reply:
ah right, well spotted! hold on